### PR TITLE
Fix exclude-na issue when patient/sample doesn't have data on selected attrs

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

**Review by kalletlak — APPROVED** [review]

File: app/scripts/controller/util.js [hunk 1]
```diff
@@ -685,6 +685,28 @@
...
+    content.strIsNa = function(str, includeEmptyStr) {
+      var status = false;
+      includeEmptyStr = _.isBoolean(includeEmptyStr) ? includeEmptyStr : false;
+      if (_.isString(str)) {
+        if (['na', 'nan', 'n/a'].indexOf(str.toLowerCase()) > -1) {
+          status = true;
+        } else if (includeEmptyStr && !str) {
+          status = true;
+        }
+      } else if (typeof str === 'undefined' || str === null) {
```
Comment (kalletlak) [inline app/scripts/controller/util.js]: not sure if `if` statement is required here

File: app/scripts/controller/util.js [hunk 2]
```diff
@@ -685,6 +685,28 @@
...
+     * @param includeEmptyStr Whether empty string should be treated as NA
+     * @returns {boolean}
+     */
+    content.strIsNa = function(str, includeEmptyStr) {
+      var status = false;
+      includeEmptyStr = _.isBoolean(includeEmptyStr) ? includeEmptyStr : false;
+      if (_.isString(str)) {
+        if (['na', 'nan', 'n/a'].indexOf(str.toLowerCase()) > -1) {
+          status = true;
+        } else if (includeEmptyStr && !str) {
```
Comment (kalletlak) [inline app/scripts/controller/util.js]: Is there a possibility for this if statement to pass?

Comment (zhx828) [reply app/scripts/controller/util.js]: Empty string

File: app/scripts/controller/util.js (hunk 1 above)
Comment (zhx828) [reply app/scripts/controller/util.js]: We need it. The first if only handles String, this one only handles undefined and null. We don't deal with Number, Object or other types

File: app/scripts/controller/util.js (hunk 2 above)
Comment (kalletlak) [reply app/scripts/controller/util.js]: instead of else if, why not or condition in above if statement

File: app/scripts/controller/util.js (hunk 1 above)
Comment (kalletlak) [reply app/scripts/controller/util.js]: isn't the input to this function is String?

Comment (zhx828) [reply app/scripts/controller/util.js]: Ideally, but could be other types

File: app/scripts/controller/util.js (hunk 2 above)
Comment (zhx828) [reply app/scripts/controller/util.js]: Cool

**Pull request merged**